Api/Controllers/EventsController.cs

```diff
@@ -74,63 +67,132 @@ public ActionResult DeleteEvent(Guid id)
-        PriceFilter priceFilter, int minCapacityFilter, int maxCapacityFilter, StartTimeFilter startTimeFilter, bool onlyAvailablePlace)
+    [HttpPost]
+    [Authorize]
+    [Route("{id}/rate")]
+    [SwaggerOperation("Rate ended event")]
+    public ActionResult<FeedbackDto> AddFeedback(Guid id, [FromBody] AddRating request)
     {
-        List<Event> filtered = events;
+        return Ok(eventService.AsUser(User.GetGuid()).AddFeedback(id, request.Rating).ToDto());
+    }
 
-        if (evNameFilter != string.Empty)
-            filtered = filtered.FindAll(e => e.Title.Contains(evNameFilter));
+    private static IEnumerable<Event> Filter(IEnumerable<Event> events, IEnumerable<TimeType>? time,
+         IEnumerable<ParticipantsType>? participants, IEnumerable<PriceType>? price,
+         string? evNameFilter, string? orgNameFilter, bool onlyAvailablePlace)
+    {
+        // Event name filter
+        if (!string.IsNullOrEmpty(evNameFilter))
+            events = events.Where(e => e.Title.Contains(evNameFilter, StringComparison.InvariantCultureIgnoreCase));
 
-        if (orgNameFilter != string.Empty)
+        // Organizer name filter
+        if (!string.IsNullOrEmpty(orgNameFilter))
         {
             var name = orgNameFilter.Split();
             string firstName = name[0];
             string lastName = string.Empty;
             if (name.Length > 1)
                 lastName = name[1];
-            filtered = filtered.FindAll(e => e.Organizer is not null && e.Organizer.FirstName.Contains(firstName) && e.Organizer.LastName.Contains(lastName));
+            events = events.Where(e => e.Organizer is not null &&
+            e.Organizer.FirstName.Contains(firstName, StringComparison.InvariantCultureIgnoreCase) &&
+            e.Organizer.LastName.Contains(lastName, StringComparison.InvariantCultureIgnoreCase));
         }
 
-        if (priceFilter == PriceFilter.Free)
-            filtered = filtered.FindAll(e => e.Fee is null);
-        else if (priceFilter == PriceFilter.Paid)
-            filtered = filtered.FindAll(e => e.Fee is not null);
-
-        filtered = filtered.FindAll(e => e.Capacity is null || (e.Capacity >= minCapacityFilter && e.Capacity <= maxCapacityFilter));
+        // Number of participants filter
+        if (participants is not null && participants.Any() && participants.Count() < 3)
+        {
+            if (participants.Count() == 1)
+            {
+                events = participants.First() switch
+                {
+                    ParticipantsType.To50 => events.Where(x => x.Participants.Count <= 50 && x.Participants.Count >= 0),
+                    ParticipantsType.From50To100 => events.Where(x => x.Participants.Count >= 50 && x.Participants.Count <= 100),
+                    ParticipantsType.Above100 => events.Where(x => x.Participants.Count >= 100),
+                    _ => throw new InvalidDomainEnumException()
+                };
+            }
```

Comment:
Is it necessarry to consider case for only one requirement separately?

Reply:
I wrote it just to work, it is not necessary but if I didn't consider it, there would be more if's below. I will simplify it in next commit.